.tree-sitter-lint/tree-sitter-lint-local/src/main.rs

```diff
@@ -0,0 +1,11 @@
+use std::sync::Arc;
+
+use tree_sitter_lint::{clap::Parser, Args, Rule};
+
+fn main() {
+    tree_sitter_lint::run_and_output(Args::parse().into_config(all_rules()));
```

Comment:
So the idea is that the custom binary hard-codes its list of "all rules"

And it gets "forwarded" the same CLI arguments that the "shell" `tree-sitter-lint` binary got invoked with

And parses those with `clap` (re-exported by `tree_sitter_lint` to make sure that versions match), turns that into a full-fledged "config" by basically attaching the rules to it, and then invoking the "main" `tree_sitter_lint` entry point (for printing output)

Reply:
I don't know if you've ever used desktop linux, but there's a window manager called XMonad that essentially does this: it provides code that you use in your config (which is code) as a library, and then knows how to rebuild itself on demand.

Reply:
Ah interesting ya not sure if I'd ever specifically heard about usages of this general "technique" before or not it just seemed like "something to reach for" when the alternatives (dynamic plugin-loading of some kind) seemed questionable

So far it's seeming kind of sane

And given that changes to one's lint config (specifically we could refine the current rebuild-triggering logic to only rebuild when the "full" set of "available" plugins/local rules changes I believe (eg if you just temporarily comment out a rule or change its configuration that shouldn't require re-building a new binary)) are relatively infrequent vs how often you actually run/use the linter it seems like probably not too "finicky" especially if we handle it pretty "automatically" (eg via these CLI/LSP server "shell" wrappers)